tools/tsp-client/src/commands.ts

```diff
@@ -242,8 +242,7 @@ export async function syncCommand(argv: any) {
     if (tspLocation.emitterPackageJsonPath) {
 
       // If the emitter package json path is provided, we need to check for the lock file in the same directory with the same name prefix.
-      const emitterPackageJsonFileName = path.basename(tspLocation.emitterPackageJsonPath, path.extname(tspLocation.emitterPackageJsonPath));
-      emitterLockPath = joinPaths(dirname(emitterPackageJsonPath), `${emitterPackageJsonFileName}-lock.json`);
+      emitterLockPath = getEmitterLockPath(emitterPackageJsonPath);
```

Comment:
Why not just always do this?   won't we always use a lock file name derived from the emitterPackageJson path, even in the default case?

Reply:
The special casing for `if (tspLocation.emitterPackageJsonPath) {` doesn't seem worth the cost of always computing the name.

Reply:
seems like I worded that oddly:)   Not special casing seems worth the cost of a string compute, even in the default case.